### PR TITLE
[OCPCLOUD-1816]: Modernize tooling

### PR DESCRIPTION
**Pull request by odvarkadaniel**
This commit introduces modernization of the approach to update auxiliary tooling to be more consistent such as in Control Plane Machine Set repository. We align envtest, controller-gen and other tooling as well as in the CPMS repo.

### CODE REVIEW

File: .ci-operator.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
 build_root_image:
   name: release
   namespace: openshift
-  tag: rhel-8-release-golang-1.18-openshift-4.11
+  tag: rhel-8-release-golang-1.19-openshift-4.11
```
Comment (JoelSpeed) [inline .ci-operator.yaml:4]: You'll need to bump this to 4.13 as well
```suggestion
  tag: rhel-8-release-golang-1.19-openshift-4.13
```

File: hack/test.sh [hunk 2]
```diff
@@ -25,8 +23,8 @@ if [ "$OPENSHIFT_CI" == "true" ] && [ -n "$ARTIFACT_DIR" ] && [ -d "$ARTIFACT_DI
 fi
 
 # Print the command we are going to run as Make would.
-echo ${GINKGO} ${GINKGO_ARGS} ${TEST_DIRS}
-${GINKGO} ${GINKGO_ARGS} ${TEST_DIRS}
+echo ${GINKGO} ${GINKGO_ARGS} ${GINKGO_EXTRA_ARGS} "<omitted>"
+${GINKGO} ${GINKGO_ARGS} ${GINKGO_EXTRA_ARGS} ./pkg/... ./cmd/...
```
Comment (JoelSpeed) [inline hack/test.sh:27]: On this one, we are probably ok with just `./...`
```suggestion
${GINKGO} ${GINKGO_ARGS} ${GINKGO_EXTRA_ARGS} ./...
```

**Comment by odvarkadaniel** [comment]
Hi @Fedosin, could you please explain whether [this](https://github.com/openshift/cluster-capi-operator/blob/main/go.mod#L250) is necessary and if so, why is it structured like that?
Thanks for explanation!

**Comment by damdo** [comment]
@odvarkadaniel [those replaces](https://github.com/openshift/cluster-capi-operator/blob/main/go.mod#L250) replace the upstream cluster-api dependency with our local fork of it. They where introduced by https://github.com/openshift/cluster-capi-operator/pull/35 and haven't been updated since. I think we can try to bump those to the latest commit of https://github.com/openshift/cluster-api

**Comment by damdo** [comment]
Hey @odvarkadaniel had to open: https://github.com/openshift/cluster-capi-operator/pull/101
for bug fixing and an overall CAPI/CAPI providers bump.
I'll take care of tooling changes too there, as go APIs needs bumping in a certain order.

**Comment by odvarkadaniel** [comment]
Closing this, as the work in this PR will be done in #101.

**Pull request closed without merging**